data/npc/lib/npc.lua

```diff
@@ -11,33 +11,34 @@ function msgcontains(message, keyword)
 end
 
 function doNpcSellItem(cid, itemid, amount, subType, ignoreCap, inBackpacks, backpack)
+	local player = Player(cid)
 	local amount = amount or 1
 	local subType = subType or 0
 	local item = 0
-	if isItemStackable(itemid) then
+	if ItemType(itemid):isStackable() then
 		if inBackpacks then
-			stuff = doCreateItemEx(backpack, 1)
-			item = doAddContainerItem(stuff, itemid, math.min(100, amount))
+			stuff = Game.createItem(backpack, 1)
+			item = doAddContainerItem(stuff, itemid, math.min(100, amount)) -- ???
 		else
-			stuff = doCreateItemEx(itemid, math.min(100, amount))
+			stuff = Game.createItem(itemid, math.min(100, amount))
 		end
-		return doPlayerAddItemEx(cid, stuff, ignoreCap) ~= RETURNVALUE_NOERROR and 0 or amount, 0
+		return player:addItemEx(stuff, ignoreCap) ~= RETURNVALUE_NOERROR and 0 or amount, 0
 	end
 
 	local a = 0
 	if inBackpacks then
-		local container, b = doCreateItemEx(backpack, 1), 1
+		local container, b = Game.createItem(backpack, 1), 1
 		for i = 1, amount do
-			local item = doAddContainerItem(container, itemid, subType)
-			if table.contains({(getContainerCapById(backpack) * b), amount}, i) then
-				if doPlayerAddItemEx(cid, container, ignoreCap) ~= RETURNVALUE_NOERROR then
+			local item = doAddContainerItem(container, itemid, subType) -- ???
```

Comment:
Same thing here as above;
"
Why did you add "-- ???"
And why did you not change the compat function? https://github.com/otland/forgottenserver/blob/master/data/lib/compat/compat.lua#L583-L594
"